src/main/java/teammates/ui/webapi/GetResponseStatsAction.java

```diff
@@ -0,0 +1,32 @@
+package teammates.ui.webapi;
+
+import teammates.common.exception.UnauthorizedAccessException;
+import teammates.common.util.Const;
+import teammates.storage.entity.FeedbackResponseRecord;
+import teammates.ui.output.FeedbackResponseRecordData;
+
+import java.util.Set;
+
+public class GetResponseStatsAction extends Action {
+    @Override
+    AuthType getMinAuthLevel() {
+        return AuthType.LOGGED_IN;
+    }
+
+    @Override
+    JsonResult execute() {
+        String durationStr = getNonNullRequestParamValue(Const.ParamsNames.DURATION);
+        String intervalStr = getNonNullRequestParamValue(Const.ParamsNames.INTERVAL);
```

Comment:
Exception handling? Both the above two lines might result in Exception and you have to try catch and return `JsonResult` of failure upon catching.

Reply:
Its handled in the ```getNonNullRequestParamValue``` method of ```Action``` class

Reply:
I see. It seems to propagate to the servlet which already handles it by returning the error in json. TIL.